deps/wazuh_testing/wazuh_testing/tools/agent_simulator.py

```diff
@@ -596,54 +671,256 @@ def get_connection_status(self):
+                        ('<random_string>', get_random_string(10)),
+                        ('<timestamp>', timestamp), ('<syscollector_type>', message_type)
+                    ]
+
+        for variable, value in fields_to_replace:
+            message = message.replace(variable, value)
+
+        message = f"{self.SYSCOLLECTOR_MQ}:{self.SYSCOLLECTOR}:{message}"
+
+        return message
+
+    def generate_event(self):
+        """Generate syscollector event.
+        
+         The event types are selected sequentially, creating a number of events of the same type specified
+         in `bath_size`.
+        """
+        if self.current_batch_events_size == 0:
+            self.current_batch_events = (self.current_batch_events + 1) % len(self.list_events)
+            self.current_batch_events_size = self.batch_size
+
+        if self.list_events[self.current_batch_events] not in ['network', 'port', 'process'] \
+                or self.current_batch_events_size > 1:
+            event = self.list_events[self.current_batch_events]
         else:
-            self.inventory_path = f"inventory/{self.os}/{self.inventory_sample}"
-        with open(self.inventory_path) as fp:
-            line = fp.readline()
-            while line:
-                if not line.startswith("#"):
-                    msg = "{0}:{1}:{2}".format(self.SYSCOLLECTOR_MQ, self.SYSCOLLECTOR, line.strip("\n"))
-                    self.inventory.append(msg)
-                line = fp.readline()
+            event = self.list_events[self.current_batch_events] + '_end'
+
+        self.current_batch_events_size = self.current_batch_events_size - 1
+        return self.format_event(event)
+
+
+class SCA:
+    """This class allows the generation of SCA events.
+
+    Create sca events, both summary and check. Example messasge:
+
+          p:sca:{"type": "check", "scan_id": 386,
+         "id": 8247694953, "policy": "CIS Benchmark for debian8",
+         "policy_id": "cis_debian8_policy", "check": {"id": 92105,
+         "title": "Ensure root is the only UID 0 account", "description": "Any account with UID 0
+         has superuser privileges on the system", "rationale": "This access must be limited to only the
+         default root account", "remediation": "Remove any users other than root with UID 0", "compliance":
+         {"cis": "6.2.6", "cis_csc": "5.1", "pci_dss": "10.2.5", "hipaa": "164.312.b", "nist_800_53":
+         "AU.14,AC.7", "gpg_13": "7.8", "gdpr_IV": "35.7,32.2", "tsc": "CC6.1,CC6.8,CC7.2,CC7.3,CC7.4"},
+         "rules": "f:/etc/passwd -> !r:^# && !r:^\\\\s*\\\\t*root: && r:^\\\\w+:\\\\w+:0:\"]",
+         "condition": "none", "file": "/etc/passwd", "result": "failed"}}
+
+    Args:
+        os (str): Agent operative system.
+    """
+    def __init__(self, os):
+        self.last_scan_id = 0
+        self.os = os
+        self.count = 0
+        self.SCA_MQ = 'p'
+        self.SCA = 'sca'
```

Comment:
It is not recommended to use upper-case variables.

Reply:
https://github.com/wazuh/wazuh-qa/commit/83483e39b469943685058b8fffef4e3ea9cd1f3b